test/tools/unified-spec-runner/match.ts

```diff
@@ -561,63 +561,65 @@ function compareEvents(
           );
         }
       }
-      return;
     } else if (expectedEvent.serverHeartbeatStartedEvent) {
```

Comment:
Why are the early returns a bug?  Aren't these all mutually exclusive?  for example - we can't expect than an event is both a heartbeat started and a heartbeat failed.

Reply:
I imagine Warren's thinking is that these early returns are redundant since we have `if/else if/else` blocks encompassing the entire function.

Reply:
It's not that they're redundant, it's that the early returns are occurring while we're looping over the array of events we want to assert against. Effectively what was happening when asserting on sdam events was that we'd exit after the first event when we had other events to check.